pkg/flavors/eksd/eksd.go

```diff
@@ -0,0 +1,113 @@
+package eksd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	distrov1alpha1 "github.com/aws/eks-distro-build-tooling/release/api/v1alpha1"
+	gerrors "github.com/pkg/errors"
+	"sigs.k8s.io/yaml"
+)
+
+// EKSD is used to wrap the eks-d manifest and provide helper functions
+type EKSD struct {
+	releaseURL string
+	release    *distrov1alpha1.Release
+}
+
+const (
+	// Flavor name of the eks-d flavor
+	Flavor string = "eks-d"
+)
+
+// New create an instance of EKSD with the manifest from the URL argument
+func New(eksdURL string) (*EKSD, error) {
+	m, err := readRelease(eksdURL)
+	if err != nil {
+		return nil, gerrors.Wrap(err, fmt.Sprintf("failed to create EKSD using manifest %s", eksdURL))
+	}
+	return &EKSD{
+		releaseURL: eksdURL,
+		release:    m,
+	}, nil
+}
+
+// KubeBinURL looks through the eksd manifest and locates the URL for the
+// binary
+func (e *EKSD) KubeBinURL(binName string) (url string, sha256 string, err error) {
+	c := e.findComponent("kubernetes")
+	if c == nil {
+		return "", "", fmt.Errorf("component %s not found in release %v", binName, e.release.Spec)
+	}
+	for _, a := range c.Assets {
+		if a.Name == "bin/linux/amd64/"+binName {
+			return a.Archive.URI, a.Archive.SHA256, nil
+		}
+	}
+	if binName == "kubeadm" {
+		return overrideKubeadm()
+	}
+	return "", "", fmt.Errorf("binary %s not found in release %v", binName, e.release.Spec)
+}
+
+func (e *EKSD) findComponent(name string) *distrov1alpha1.Component {
+	n := strings.ToLower(name)
+	for _, c := range e.release.Status.Components {
+		if strings.ToLower(c.Name) == n {
+			return &c
+		}
+	}
+	return nil
+
+}
+func overrideKubeadm() (string, string, error) {
```

Comment:
Please add an issue to track this so we can replace this when upstream is fixed.

Reply:
Done